### PR TITLE
Removing and syncing duplicate configs

### PR DESCRIPTION
**Pull request by NaMNDV**

### CODE REVIEW

**Review by dargudear-google — APPROVED** [review]

File: deploy/provider-gcp-plugin.yaml [hunk 1]
```diff
@@ -75,8 +75,8 @@ spec:
         - name: provider
           image: us-docker.pkg.dev/secretmanager-csi/secrets-store-csi-driver-provider-gcp/plugin@sha256:bf97decbbd5b5894662c438b6720bc3e42815301a507f5a52bd75771c0488cb6
           securityContext:
-            runAsUser: 1000
-            runAsGroup: 1000
+            runAsUser: 0
```
Comment (amitmodak) [inline deploy/provider-gcp-plugin.yaml:78]: Can you please explain why is this change required?

Comment (NaMNDV) [reply deploy/provider-gcp-plugin.yaml:78]: The driver has a requirement to [run it as root](https://secrets-store-csi-driver.sigs.k8s.io/concepts.html#security:~:text=The%20provider%20plugins%20are%20also%20required%20to%20run%20as%20root%20(though%20privileged%20should%20not%20be%20necessary).%20This%20is%20because%20the%20provider%20plugin%20must%20create%20a%20unix%20domain%20socket%20in%20a%20hostPath%20for%20the%20driver%20to%20connect%20to). The provider server won't turnup otherwise 

`
{"ts":1699006552784.2349,"caller":"secrets-store-csi-driver-provider-gcp/main.go:186","msg":"unable to listen to unix socket","path":"/etc/kubernetes/secrets-store-csi-providers/gcp.sock","err":"listen unix /etc/kubernetes/secrets-store-csi-providers/gcp.sock: bind: permission denied"}
`

Comment (amitmodak) [reply deploy/provider-gcp-plugin.yaml:78]: Sorry, I don't get it. The current version on main and release has this value set to 1000 and it seems to be working. Can you please elaborate the exact issue?

Comment (NaMNDV) [reply deploy/provider-gcp-plugin.yaml:78]: This config was working before because there was a duplication of config and the config with our changes was overridden by other config: https://github.com/GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp/issues/340, while fixing the issue and removing the duplicates, we found out that the pods are not turning up and in further investigation, we found out that the driver had a requirement to run as a root to obtain certain accesses. Thus removing the requirement and using default as 0
